distributed/tensor_parallelism/sequence_parallel_example.py

```diff
@@ -33,51 +39,75 @@
 """
 
 
-def demo_sp(rank, args):
+def demo_sp(args):
     """
     Main body of the demo of a basic version of sequence parallel by using
     PyTorch native APIs.
     """
-    print(f"Running SP example on rank {rank}.")
-    setup(rank, args.world_size)
+    torchrun_setup()
+
+    # understand world topology
+    _rank = int(os.environ["RANK"])
+    _local_rank = int(os.environ["LOCAL_RANK"])
+    _world_size = int(os.environ["WORLD_SIZE"])
+    _local_world_size = int(os.environ["LOCAL_WORLD_SIZE"])
+
+    torch.cuda.set_device(_local_rank)
+
+    def rank_print(msg):
+        """helper function to print only on global rank 0"""
+        if _rank==0:
+            print(f"{msg}")
+
+    print(f"Running basic Megatron style Sequence Parallel example on rank {_rank}.")
+
+    # create a device mesh based on the given world_size.
+
+    device = f"cuda"
+    device_mesh = init_device_mesh(device_type = device,mesh_shape = (_world_size,))
+    assert device_mesh is not None, "unable to create valid device mesh"
+
+    rank_print(f"Device Mesh created: {device_mesh=}")
```

Comment:
hmm I feel this can be a logger.debug instead

Reply:
everything moved to logger now.